pysteps/nowcasts/steps.py

```diff
@@ -33,6 +33,863 @@
+        self.kmperpixel = kwargs.get("kmperpixel", None)
+        self.timestep = kwargs.get("timestep", None)
+        self.extrap_method = kwargs.get("extrap_method", "semilagrangian")
+        self.decomp_method = kwargs.get("decomp_method", "fft")
+        self.bandpass_filter_method = kwargs.get("bandpass_filter_method", "gaussian")
+        self.noise_method = kwargs.get("noise_method", "nonparametric")
+        self.noise_stddev_adj = kwargs.get("noise_stddev_adj", None)
+        self.ar_order = kwargs.get("ar_order", 2)
+        self.vel_pert_method = kwargs.get("vel_pert_method", "bps")
+        self.conditional = kwargs.get("conditional", False)
+        self.probmatching_method = kwargs.get("probmatching_method", "cdf")
+        self.mask_method = kwargs.get("mask_method", "incremental")
+        self.seed = kwargs.get("seed", None)
+        self.num_workers = kwargs.get("num_workers", 1)
+        self.fft_method = kwargs.get("fft_method", "numpy")
+        self.domain = kwargs.get("domain", "spatial")
+        self.extrap_kwargs = kwargs.get("extrap_kwargs", None)
+        self.filter_kwargs = kwargs.get("filter_kwargs", None)
+        self.noise_kwargs = kwargs.get("noise_kwargs", None)
+        self.vel_pert_kwargs = kwargs.get("vel_pert_kwargs", None)
+        self.mask_kwargs = kwargs.get("mask_kwargs", None)
+        self.measure_time = kwargs.get("measure_time", False)
+        self.callback = kwargs.get("callback", None)
+        self.return_output = kwargs.get("return_output", True)
+
+        # Additional variables for internal state management
+        self.fft = None
+        self.bp_filter = None
+        self.extrapolator_method = None
+        self.domain_mask = None
+        self.precip_cascades = None
+        self.gamma = None
+        self.phi = None
+        self.pert_gen = None
+        self.noise_std_coeffs = None
+        self.randgen_prec = None
+        self.randgen_motion = None
+        self.velocity_perturbations = None
+        self.precip_forecast = None
+        self.mask_prec = None
+        self.mask_thr = None
+        self.precip_decomp = None
+        self.vp_par = None
+        self.vp_perp = None
+        self.fft_objs = None
+        self.generate_noise = None
+
+        # Additional variables for time measurement
+        self.start_time_init = None
+        self.init_time = None
+        self.mainloop_time = None
+
+        # Initialize number of ensemble workers
+        self.num_ensemble_workers = min(self.n_ens_members, self.num_workers)
+
+    def compute_forecast(self):
+        """
+        Main loop for nowcast ensemble generation. This handles extrapolation,
+        noise application, autoregressive modeling, and recomposition of cascades.
+        """
+        self._check_inputs()
+        self._print_forecast_info()
+        # Measure time for initialization
+        if self.measure_time:
+            self.start_time_init = time.time()
```

Comment:
Should we add a test where measure_time is True (and False)? That would probably tick the box for the test coverage.

Reply:
We could to this but I don't know if it is worth it, its an internal method with almost no functionality...